backend/src/controllers/userController.js

```diff
@@ -42,6 +42,22 @@ export default class UserController {
     });
   }
 
+  // check if user is already in the database or not
+  checkUser(req) {
+    return new Promise((resolve, reject) => {
+      const UserModel = new User();
+      UserModel.getOneUser(req.body.email, (err, res) => {
+        if (err) {
+          reject({ error: err });
+        }
+        if (res && res.length > 0 && res[0].email === req.body.email) {
+          reject({ error: "Email already exists" });
+        }
+        resolve(res);
+      });
+    });
+  }
+
```

Comment:
Wasn't really sure how to connect endpoint to frontend part of SignUpForm

Reply:
What I would do is make an endpoint that is a get, which takes in a param as a user email. And then you'll basically just run a SELECT * from users where user_email = <EMAIL>; and basically if the result is not 0, the endpoint can return a false and that will ultimately fail the signup.

Or, you can do the check directly in the POST user endpoint, and in the controller for that, do a check to see if the user exists, and if so, fail that POST signup user endpoint.

Reply:
If I go with the first option, do I need to modify the usersProc.sql file? I also noticed there's loadSingleUser which has SELECT * FROM users WHERE email = _email;

Reply:
oh yeah then I think you can just reuse that one, and not have to deal with the database, you can use that model call as a part of the signup process and do the check there